src/vstest.console/CommandLine/CommandLineOptions.cs

```diff
@@ -283,28 +284,33 @@ public void AddSource(string source)
             {
                 throw new CommandLineException(CommandLineResources.CannotBeNullOrEmpty);
             }
-
+            var filePatternParser = new FilePatternParser();
             source = source.Trim();
-
+            
             // Convert the relative path to absolute path
-            if(!Path.IsPathRooted(source))
+            if (!Path.IsPathRooted(source))
             {
                 source = Path.Combine(FileHelper.GetCurrentDirectory(), source);
             }
 
-            if (!FileHelper.Exists(source))
-            {
-                throw new CommandLineException(
-                    string.Format(CultureInfo.CurrentUICulture, CommandLineResources.TestSourceFileNotFound, source));
-            }
+            var sourceFiles = filePatternParser.GetMatchingFiles(source);
 
-            if (this.sources.Contains(source, StringComparer.OrdinalIgnoreCase))
+            foreach (var sourceFile in sourceFiles)
             {
-                throw new CommandLineException(
-                    string.Format(CultureInfo.CurrentCulture, CommandLineResources.DuplicateSource, source));
-            }
+                if (!FileHelper.Exists(sourceFile))
+                {
+                    throw new CommandLineException(
+                        string.Format(CultureInfo.CurrentUICulture, CommandLineResources.TestSourceFileNotFound, source));
+                }
 
-            this.sources.Add(source);
+                if (this.sources.Contains(sourceFile, StringComparer.OrdinalIgnoreCase))
```

Comment:
We throw an error if there are duplicate files given in commandline. With wildcard support, this may increase because same file might get picked up due to different patterns given. Should we make this a warning instead of error?
@singhsarab @PBoraMSFT

Reply:
Also, at the end of going through all wildcard filenames, should we print the list of files selected on the console?